src/ddsrt/include/dds/ddsrt/cdtors.h

```diff
@@ -14,16 +14,40 @@
 #include "dds/export.h"
 #include "dds/ddsrt/sync.h"
 
+/**
+ * @file cdtors.h
+ * @brief Contains initializing and finalizing functions for some ddsrt units, and provides a singleton mutex and cond.
+ * 
+ * Initializes and finalizes mutexes of ddsrt/random and ddsrt/atomics, and for Windows also the ddsrt/socket
+ * (not necessary for other platforms).
+ */
+
 #if defined (__cplusplus)
 extern "C" {
 #endif
 
+/**
+ * @brief Initialize ddsrt units
```

Comment:
I think it needs two extra bit of information:
* I counts the number of invocations of `ddsrt_init`, and only the "first" (scare quotes because of `ddsrt_fini`) actually initializes it.
* Any concurrent attempts at initializing it will wait until it has been initialized.